scripts/genVis.py

```diff
@@ -1,58 +1,171 @@
+__author__ = 'Jon Mallen'
+
+import os
+import sys
 from sklearn.manifold import TSNE
-#from MulticoreTSNE import MulticoreTSNE as TSNE
 from sklearn.decomposition import PCA
-import numpy as np
+from umap import UMAP
 import matplotlib.pyplot as plt
-import matplotlib.cm as cm
 import pandas as pd
-from optparse import OptionParser 
-parser = OptionParser()
+import argparse
+import itertools
 
-parser.add_option('-i','--inFile',default='',type=str,
-                  help='Specify input expression matrix file name')
+# Define arguments
+parser = argparse.ArgumentParser("Visualize the simulated single-cell gene expression data output by BoolODE.")
+parser.add_argument('-f', '--pathToFiles', default='', type=str, help='Specify path to ExpressionData.csv and '
+                                                                      'PseudoTime.csv files generated by the BoolODE '
+                                                                      'simulation, as well as the ClusterIds.csv '
+                                                                      'present if the user specified at least two '
+                                                                      'clusters in the simulation.')
+parser.add_argument('-p', '--pca', nargs='*', help='Use PCA for visualizing the data. '
+                                                   'Specify the number of dimensions (2 or 3) as argument. '
+                                                   'Default is 2.')
+parser.add_argument('-t', '--tsne', nargs='*', help='Use t-SNE for visualizing the data. '
+                                                    'Specify the number of dimensions (2 or 3) as argument. '
+                                                    'Default is 2.')
+parser.add_argument('-u', '--umap', nargs='*', help='Use UMAP for visualizing the data. '
+                                                    'Specify the number of dimensions (2 or 3) as argument. '
+                                                    'Default is 2.')
+parser.add_argument('-c', '--clusterFile', action='store_true', default=False,
+                    help='Use the cluster file ClusterIds.csv to assign clusters if the user specified at least two '
+                         'clusters in the simulation.')
+parser.add_argument('-n', '--plotName', default='', nargs='*', help='Name of the plot.')
```

Comment:
Do you think we should instead allow the user to customize the plot name for each of the plots generated? As it is right now, passing the `--plotName` argument would give the same plot name for each of the plots. Another alternative would be to simply fix the plot title for each to something reasonable based on the method of dimensional reduction performed. I will go with whatever you think is best.